stage1_fly/run/main.go

```diff
@@ -0,0 +1,391 @@
+	err = ioutil.WriteFile(filepath.Join(out, "ppid"),
+		[]byte(fmt.Sprintf("%d\n", pid)), 0644)
+	if err != nil {
+		return fmt.Errorf("Cannot write ppid file: %v\n", err)
+	}
+	return nil
+}
+
+func evaluateMounts(rfs string, app string, p *stage1commontypes.Pod) ([]flyMount, error) {
+	imApp := p.Images[app].App
+	namedVolumeMounts := map[types.ACName]volumeMountTuple{}
+
+	for _, m := range p.Manifest.Apps[0].Mounts {
+		_, exists := namedVolumeMounts[m.Volume]
+		if exists {
+			log.Fatalf("fly: duplicated mount given: %q", m.Volume)
+		}
+		namedVolumeMounts[m.Volume] = volumeMountTuple{M: m}
+		log.Printf("Adding %+v", namedVolumeMounts[m.Volume])
+	}
+
+	// Merge command-line Mounts with ImageManifest's MountPoints
+	for _, mp := range imApp.MountPoints {
+		tuple, exists := namedVolumeMounts[mp.Name]
+		switch {
+		case exists && tuple.M.Path != mp.Path:
+			return nil, fmt.Errorf("fly: conflicting path information from mount and mountpoint %q", mp.Name)
+		case !exists:
+			namedVolumeMounts[mp.Name] = volumeMountTuple{M: schema.Mount{Volume: mp.Name, Path: mp.Path}}
+			log.Printf("Adding %+v", namedVolumeMounts[mp.Name])
+		}
+	}
+
+	// Insert the command-line Volumes
+	for _, v := range p.Manifest.Volumes {
+		// Check if we have a mount for this volume
+		tuple, exists := namedVolumeMounts[v.Name]
+		if !exists {
+			return nil, fmt.Errorf("fly: missing mount for volume %q", v.Name)
+		} else if tuple.M.Volume != v.Name {
+			// TODO(steveeJ): remove this case. it's merely a safety mechanism regarding the implementation
+			return nil, fmt.Errorf("fly: mismatched volume:mount pair: %q != %q", v.Name, tuple.M.Volume)
+		}
+		namedVolumeMounts[v.Name] = volumeMountTuple{V: v, M: tuple.M}
+		log.Printf("Adding %+v", namedVolumeMounts[v.Name])
+	}
+
+	// Merge command-line Volumes with ImageManifest's MountPoints
+	for _, mp := range imApp.MountPoints {
+		// Check if we have a volume for this mountpoint
+		tuple, exists := namedVolumeMounts[mp.Name]
+		if !exists || tuple.V.Name == "" {
+			log.Fatalf("fly: missing volume for mountpoint %q", mp.Name)
+		}
+
+		// If empty, fill in ReadOnly bit
+		if tuple.V.ReadOnly == nil {
+			v := tuple.V
+			v.ReadOnly = &mp.ReadOnly
+			namedVolumeMounts[mp.Name] = volumeMountTuple{M: tuple.M, V: v}
+			log.Printf("Adding %+v", namedVolumeMounts[mp.Name])
+		}
+	}
+
+	// Gather host mounts which we make MS_SHARED if passed as a volume source
+	hostMounts, err := getHostMounts()
+	if err != nil {
+		log.Fatalf("fly: error gathering host mounts")
+	}
+
+	argFlyMounts := []flyMount{}
+	var flags uintptr = syscall.MS_BIND // TODO: make this optional: | syscall.MS_REC
+	for _, tuple := range namedVolumeMounts {
+		if _, isHostMount := hostMounts[tuple.V.Source]; isHostMount {
+			// Mark the host mount as SHARED so the container's changes to the mount are propagated to the host
+			argFlyMounts = append(argFlyMounts,
+				flyMount{"", "", tuple.V.Source, "none", syscall.MS_REC | syscall.MS_SHARED},
```

Comment:
Yes, but it's not for the actual bind-mount. It's for making the host-side of the bind-mount RSHARED if it's a mount itself.